File: archive/2019/solutions/BakingPan.py
n = int(input())

minx = 20000001
maxx = -20000001
miny = 20000001
maxy = -20000001
for i in range(n):
    st = [int(x) for x in input().split()]
    minx = min(minx,st[0]-st[2])
    maxx = max(maxx,st[0]+st[2])
    miny = min(miny,st[1]-st[2])
    maxy = max(maxy,st[1]+st[2])

print((maxx-minx)*(maxy-miny))

File: archive/2019/solutions/BakingPan.java

import java.util.*;
import java.io.*;

// By Ayush Varshney
public class BakingPan {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long minX = Long.MAX_VALUE, maxX = Long.MIN_VALUE, minY = Long.MAX_VALUE, maxY = Long.MIN_VALUE;
        while (n-- > 0) {
            int x = sc.nextInt(), y = sc.nextInt(), r = sc.nextInt();
            minX = Math.min(minX, x-r);
            maxX = Math.max(maxX, x+r);
            minY = Math.min(minY, y-r);
            maxY = Math.max(maxY, y+r);
        }
        long area = (maxX-minX)*(maxY-minY);
        System.out.println(area);
        sc.close();
    }
    
}